src/types/thread_based_random_provider.rs

```diff
@@ -0,0 +1,50 @@
+use rand::Rng;
+use crate::models::generation_base::GenerationBase;
+use crate::interfaces::random_provider::RandomProvider;
+
+pub struct ThreadBasedRandomProvider {
+}
```

Comment:
Empty struct defnitions should be in one line.
```suggestion
pub struct ThreadBasedRandomProvider {}
```